server/main.go

```diff
@@ -72,30 +71,12 @@ func (p *Plugin) OnActivate() error {
 
 	potentiallyUpdatedConfig, wasUpdated, err := runAllMigrations(p.API, pluginAPI, *p.configuration.Config())
 	if err != nil {
-		// Log error but don't fail activation - migrations may have already run on another instance
-		pluginAPI.Log.Error("failed to run migrations", "error", err)
-	} else if wasUpdated {
-		// Wrap config in the configuration struct that has the proper nesting
-		wrappedConfig := configuration{Config: potentiallyUpdatedConfig}
-
-		// Convert config to map[string]any for plugin API
-		out := map[string]any{}
-		marshalBytes, marshalErr := json.Marshal(wrappedConfig)
-		if marshalErr != nil {
-			return fmt.Errorf("failed to marshal migrated configuration: %w", marshalErr)
-		}
-		if unmarshalErr := json.Unmarshal(marshalBytes, &out); unmarshalErr != nil {
-			return fmt.Errorf("failed to unmarshal migrated configuration: %w", unmarshalErr)
-		}
-
-		// Persist to database via API
-		if saveErr := pluginAPI.Configuration.SavePluginConfig(out); saveErr != nil {
-			return fmt.Errorf("failed to save migrated configuration: %w", saveErr)
-		}
-
+		return fmt.Errorf("failed to run migrations: %w", err)
```

Comment:
I changed this to an explicit failure if we can't run the migration. If it fails and we end up in a bad state, or if we're just using the old config structure, the plugin won't work at all.